zerostash/src/commands/mount.rs

```diff
@@ -0,0 +1,279 @@
+
+    // Ensure the filesystem is unmounted.
+    handle.join();
+
+    Ok(())
+}
+
+type DirPath = PathBuf;
+type FilePath = PathBuf;
+type FileContent = Vec<u8>;
+type Metadata = Arc<Entry>;
+
+struct SimpleFs {
+    destroy_tx: mpsc::SyncSender<()>,
+    stash: Infinitree<Files>,
+    file_map: HashMap<FilePath, Metadata>,
+    file_content: Mutex<HashMap<FilePath, FileContent>>,
+    dir_map: HashMap<DirPath, Vec<DirectoryEntry>>,
+    file_attr_memory: Mutex<HashMap<PathBuf, FileAttr>>,
+}
+
+fn add_dir_to_map(
+    dir_map: &mut HashMap<DirPath, Vec<DirectoryEntry>>,
+    path: &Path,
+    kind: FileType,
+) {
+    let name = path
+        .file_name()
+        .expect("All files have filenames")
+        .to_owned();
+
+    let parent = path
+        .parent()
+        .expect("Paths should have parents")
+        .to_path_buf();
+
+    if !dir_map.contains_key(&parent) {
+        add_dir_to_map(dir_map, &parent, FileType::Directory);
+    }
+
+    dir_map
+        .entry(parent)
+        .or_default()
+        .push(DirectoryEntry { name, kind });
+}
+
+const DIR_ATTR: FileAttr = FileAttr {
+    size: 0,
+    blocks: 0,
+    atime: SystemTime::UNIX_EPOCH,
+    mtime: SystemTime::UNIX_EPOCH,
+    ctime: SystemTime::UNIX_EPOCH,
+    crtime: SystemTime::UNIX_EPOCH,
+    kind: FileType::Directory,
+    perm: 0o444,
+    nlink: 1,
+    uid: 1000,
+    gid: 1000,
+    rdev: 0,
+    flags: 0,
+};
+
+pub fn match_filetype(file_type: &zerostash_files::FileType) -> FileType {
+    match file_type {
+        zerostash_files::FileType::File => FileType::RegularFile,
+        zerostash_files::FileType::Symlink(_) => FileType::Symlink,
+        zerostash_files::FileType::Directory => panic!("Didnt expect a directory!"),
+    }
+}
+
+impl SimpleFs {
+    pub fn open(
+        stash: Infinitree<Files>,
+        options: &restore::Options,
+        destroy_tx: mpsc::SyncSender<()>,
+    ) -> Result<Self> {
+        let mut dir_map: HashMap<DirPath, Vec<DirectoryEntry>> = HashMap::new();
+        dir_map.insert(PathBuf::new(), vec![]);
+        let mut file_map = HashMap::new();
+
+        for file in options.list(&stash) {
+            let file_name_path = PathBuf::from(&file.name);
+            let filetype = match_filetype(&file.file_type);
+
+            add_dir_to_map(&mut dir_map, &file_name_path, filetype);
+            file_map.insert(file_name_path, file.clone());
+        }
+
+        Ok(SimpleFs {
+            destroy_tx,
+            stash,
+            file_map,
+            file_content: Mutex::new(HashMap::new()),
+            dir_map,
+            file_attr_memory: Mutex::new(HashMap::new()),
+        })
+    }
+}
+
+const TTL: Duration = Duration::from_secs(1);
+
+pub fn file_to_fuse(file: &Arc<Entry>) -> FileAttr {
+    FileAttr {
+        size: file.size,
+        blocks: 1,
+        atime: SystemTime::now(),
+        mtime: SystemTime::now(),
+        ctime: SystemTime::now(),
```

Comment:
These are actually all inside `Entry`, so we can map them, like you do with `gid` and `uid`.

Reply:
Is `atime` really stored? From what I see `unix_secs` and `unix_nanos` is constructed from `mtime`.

Reply:
Ah, you're correct. We only store `mtime`, not the rest. `atime` could probably be the 0s commit timestamp that we're looking at.